From 69c9fe99c03d2df76b888e918945031605be5992 Mon Sep 17 00:00:00 2001
From: Lucas Dietrich <lucas.dietrich.git@proton.me>
Date: Wed, 3 Jun 2026 19:38:50 +0200
Subject: [PATCH] stm32mp257-dk-amy: add custom devicetree

* define aliases for MMC devices
* enable usart6 (requires -DTRACE=OFF in USBPD_DRP_UCSI demo firmware)

Upstream-Status: Inappropriate [board-specific devicetree, not for upstream]
---
 arch/arm64/boot/dts/st/Makefile               |  1 +
 arch/arm64/boot/dts/st/stm32mp257f-dk-amy.dts | 27 +++++++++++++++++++
 2 files changed, 28 insertions(+)
 create mode 100644 arch/arm64/boot/dts/st/stm32mp257f-dk-amy.dts

diff --git a/arch/arm64/boot/dts/st/Makefile b/arch/arm64/boot/dts/st/Makefile
index 63908113a..a57327cae 100644
--- a/arch/arm64/boot/dts/st/Makefile
+++ b/arch/arm64/boot/dts/st/Makefile
@@ -3,4 +3,5 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp215f-dk.dtb \
 	stm32mp235f-dk.dtb \
 	stm32mp257f-dk.dtb \
+	stm32mp257f-dk-amy.dtb \
 	stm32mp257f-ev1.dtb
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk-amy.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk-amy.dts
new file mode 100644
index 000000000..9c164379c
--- /dev/null
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk-amy.dts
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2026 - All Rights Reserved
+ */
+
+/dts-v1/;
+
+#include "stm32mp257f-dk.dts"
+
+/ {
+	model = "STMicroelectronics STM32MP257F-DK Discovery Board (Amy Board)";
+
+	aliases {
+		mmc0 = &sdmmc1; // Force SD CARD as mmc0
+		mmc1 = &sdmmc2; // Force eMMC as mmc1
+	};
+
+	gpio-keys {
+		button_user_2: button-user-2 {
+			status = "okay";
+		};
+	};
+};
+
+&usart6 {
+	status = "okay";
+};
\ No newline at end of file
-- 
2.55.0

